src/main/java/nl/knaw/dans/dataverse/tools/datasetmover/dvnconnect/DvnApiConnector.java

```diff
@@ -93,35 +97,52 @@ private String getApiToken(String username, String password) {
         String[] hdl = dataset.split("/");
-        //http://ddvn.dans.knaw.nl:8080/api/datasets/:persistentId/?persistentId=hdl:10411/10021
         JsonObject jo = getResponseAsJsonObject(HOST + "datasets/:persistentId/?persistentId=hdl%3A" + hdl[0] + "%2F" + hdl[1] + "&key=" + apiToken);
         if (jo != null && jo.getString("status").equals("OK"))
             return jo.getJsonObject("data").getInt("id");
-
+        LOG.warn("No dataset found. Dataset: " + dataset);
         return -1;
     }
 
     public int findDataverseIdByAlias(String alias) {
-        //http://ddvn.dans.knaw.nl:8080/api/dataverses/4tu?key=c7c57eaa-a133-4f2d-976a-5a40e69e7435
         JsonObject jo = getResponseAsJsonObject(HOST + "dataverses/" + alias + "?key=" + apiToken);
         if (jo != null && jo.getString("status").equals("OK"))
             return jo.getJsonObject("data").getInt("id");
 
         return -1;
     }
 
-    public boolean updateDatasetOwner(String dataverseAlias, int datasetId, String persistentId) {
-        //Unfortunately, there is no api for updating dataset owner
-        DvobjectBean dob = new DvobjectBean();
-        boolean succes = dob.updateOwnerId(findDataverseIdByAlias(dataverseAlias), datasetId);
-        if (succes) {
-            //run index
-            //curl http://localhost:8080/api/admin/index/dataset?persistentId=hdl:10411/YOSOSF
-            String[] hdl = persistentId.split("/");
-            JsonObject jo = getResponseAsJsonObject(HOST + "admin/index/dataset?persistentId=hdl%3A" + hdl[0] + "%2F" + hdl[1]);
-            succes = (jo != null && jo.getString("status").equals("OK"));
+    public boolean updateDatasetOwner(int datasetId, String persistentId, String targetDataverseAlias) {
+
+        JsonObject jsonObject = null;
+        try {
+            URL obj = new URL(HOST + "datasets/"+ datasetId + "/moveTo/" + targetDataverseAlias + "?key=" + apiToken);
+            HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+
+            con.setRequestMethod("POST");
+            con.setDoOutput(true);
+            if (con.getResponseCode() == HttpURLConnection.HTTP_OK) {
+                JsonReader reader = Json.createReader(
+                        new StringReader(IOUtils.toString(con.getInputStream(), "UTF-8")));
+
+                jsonObject = reader.readObject();
+                reader.close();
+
+                if (jsonObject != null && jsonObject.getString("status").equals("OK")) {
+                    //run index
+                    String[] hdl = persistentId.split("/");
+                    JsonObject joi = getResponseAsJsonObject(HOST + "admin/index/dataset?persistentId=hdl%3A" + hdl[0] + "%2F" + hdl[1]);
+                    return (joi != null && joi.getString("status").equals("OK"));
+                }
+                return false;
+            }
+
+
+        } catch (IOException e) {
+            LOG.error("IOException, message: " + e.getMessage());
```

Comment:
About the log and error messages; tell what the attempt was that failed, just echoing the name of the exceptions does not add information.